### DISCUSSION TITLE
How to automatically regenerate project when solution file changes in visual studio? Something equivelent to cmake's ZERO_CHECK.

### DISCUSSION THREAD

**Discussion by speakfool** [premake/premake-core#2151] [General]
Hi, maybe it's a bit dumb, but I cannot find related answers.

Currently I need to manaully invoke the premake.exe when project changes. How can automatic do this? In CMake, there's a ZERO_CHECK project that does the stuff, how is this done in premake? 

Thanks!

**Comment by speakfool** [comment]
Oh, I just saw it's on the list for premake6 in this link. Is there anything we could try now?

https://premake.github.io/blog/2021/11/21/community-update-10

**Reply by learn-more** [reply]
There seems to be something along the lines of what you are asking:
https://premake.github.io/docs/editorintegration/

**Reply by speakfool** [reply]
Thanks! Turns out it's too heavy for me and team. 
We end up with adding an External Tool in Visual Studio, with a custom .bat commandline that runs premake manually.
Then assign it with a shortcut, solved partial of the problem.
